feat: delete unused support bundle after a period of time

Yu-Jack opened the pull request:
**Problem:**

Support bundle CR isn't deleted due to refreshing window or something which terminates the downloading process. The leftover support bundle manager occupies the cluster resource.

**Solution:**

***We choose 30 minutes in the end***

Check the exist time with `SupportBundleInitialized` status (This status is in final state).

if this time is longer than 10 minutes, that means it might be broken or error, then delete the support bundle CR. 

I think 10 minutes is enough to wait after the support bundle is ready for downloading.

**Related Issue:**

https://github.com/harvester/harvester/issues/4922

**Test plan:**

1. Create support bundle in harvester dashboard.
2. Refresh the window to avoid the harvester dashboard to trigger download API.
3. Support bundle CR should be deleted after x seconds.
  Go to settings to change `support-bundle-expiration` to like 1 (mean 1 minute), support bundle will be delete in 1 minute. Then, use `k get supportbundles -n harvester-system` to check resource.

Yu-Jack committed:
feat: delete support bundle if it exists for a while

Signed-off-by: Jack Yu <jack.yu@suse.com>

Yu-Jack committed:
test: add test cases

Signed-off-by: Jack Yu <jack.yu@suse.com>

Yu-Jack committed:
feat: use configurable setting instead of const value

Signed-off-by: Jack Yu <jack.yu@suse.com>

Yu-Jack committed:
refactor: renaming

Signed-off-by: Jack Yu <jack.yu@suse.com>

bk201 reviewed (approved):
LGTM, thanks!

w13915984028 reviewed (commented):
Please check those robustness related questions, thanks.

Yu-Jack committed:
feat: add following change by feedback

1. add webhook for support bundle expiration seconds
2. ignore not found error while deleting support bundle
3. change setting GetInt logic to skip one step if value is empty string
4. use helper function wranger provides to get condition

Signed-off-by: Jack Yu <jack.yu@suse.com>

Yu-Jack committed:
refactor: use minute instead of seconds

Signed-off-by: Jack Yu <jack.yu@suse.com>

Yu-Jack committed:
fix: assign supportbundlexpiration of default value in controller if setting return 0 accidentally

Signed-off-by: Jack Yu <jack.yu@suse.com>

Yu-Jack committed:
refactor: move out the constant to avoud import cycles

Signed-off-by: Jack Yu <jack.yu@suse.com>

w13915984028 reviewed (approved):
LGTM, thanks.

btw, please squash the commits before final merge.

Yu-Jack committed:
fix: use provided function and change requeuing time

Signed-off-by: Jack Yu <jack.yu@suse.com>

ibrokethecloud reviewed (approved):
lgtm. thanks.
